KLT-tracker_class.py

```diff
@@ -0,0 +1,189 @@
+                                   blockSize=self.block_size)
+
+    def detect_features(self, old_frame):
+        old_gray = cv.cvtColor(old_frame, cv.COLOR_BGR2GRAY)
+        p0 = cv.goodFeaturesToTrack(old_gray, mask=None, **self.feature_params)
+        return p0
+
+    def find_good(self, img1, shot):
+        MIN_MATCH_COUNT = 10
+        sift = cv.SIFT_create()
+        kp1, des1 = sift.detectAndCompute(img1, None)
+        kp2, des2 = sift.detectAndCompute(shot, None)
+        FLANN_INDEX_KDTREE = 1
+        index_params = dict(algorithm=FLANN_INDEX_KDTREE, trees=5)
+        search_params = dict(checks=50)
+        flann = cv.FlannBasedMatcher(index_params, search_params)
+
+        matches = flann.knnMatch(des1, des2, k=2)
+
+        good = []
+        for m, n in matches:
+            if m.distance < 0.6 * n.distance:
+                good.append(m)
+
+        if len(good) > MIN_MATCH_COUNT:
+            src_pts = np.float32([kp1[m.queryIdx].pt for m in good]).reshape(-1, 1, 2)
+            dst_pts = np.float32([kp2[m.trainIdx].pt for m in good]).reshape(-1, 1, 2)
+
+            M, mask = cv.findHomography(src_pts, dst_pts, cv.RANSAC, 5.0)
+            matchesMask = mask.ravel().tolist()
+
+            h, w = img1.shape
+            pts = np.float32([[0, 0], [0, h - 1], [w - 1, h - 1], [w - 1, 0]]).reshape(-1, 1, 2)
+            dst = cv.perspectiveTransform(pts, M)
+            features = []
+
+            ob = []
+            for i, m in enumerate(good):
+                if matchesMask[i]:
+                    features.append(dst_pts[i])
+                    ob.append(src_pts[i])
+
+            s_d = []
+            for i in ob:
+                x, y = i[0][0], i[0][1]
+                s_d.append([x, y, 0])
+            s_d = np.array(s_d)
+
+            features = np.array(features)
+
+        else:
+            print("Not enough matches are found - {}/{}".format(len(good), MIN_MATCH_COUNT))
+            matchesMask = None
+
+        return dst, features, s_d
+
+    def track_features(self, old_frame, frame, p0):
+        frame_gray = cv.cvtColor(frame, cv.COLOR_BGR2GRAY)
+        old_gray = cv.cvtColor(old_frame, cv.COLOR_BGR2GRAY)
+
+        p1, st, err = cv.calcOpticalFlowPyrLK(old_gray, frame_gray, p0, None, **self.lk_params)
+
+        good_new = p1[st == 1]
+        good_old = p0[st == 1]
+
+        return good_new, good_old
+
+    def track_features_sift(self, old_frame, frame, feat, flag):
+        frame_gray = cv.cvtColor(frame, cv.COLOR_BGR2GRAY)
+        old_gray = cv.cvtColor(old_frame, cv.COLOR_BGR2GRAY)
+
+        p_feat, st_feat, err_feat = cv.calcOpticalFlowPyrLK(old_gray, frame_gray, feat, None, **self.lk_params)
+
+        good_new_feat = p_feat[st_feat == 1]
+        good_old_feat = feat[st_feat == 1]
+
+        if flag:
+            return good_new_feat, good_old_feat, p_feat
+
+        return good_new_feat, good_old_feat
+
+    def draw_tracks(self, mask, frame, good_new, good_old):
```

Comment:
Drawing functions do not apply to the algorithm, suggest to move them to visualizer class or separate functions